### PR TITLE
add misc puppet-lint plugins

### PR DESCRIPTION
**Pull request by jhoblitt**

### CODE REVIEW

**Comment by jhoblitt** [comment]
I started converting the legacy facts but got tired of fighting the tests...

File: Gemfile [hunk 1]
```diff
@@ -28,6 +28,14 @@ group :development do
   gem "puppet-module-posix-dev-r#{minor_version}", '~> 0.4',     require: false, platforms: [:ruby]
   gem "puppet-module-win-default-r#{minor_version}", '~> 0.4',   require: false, platforms: [:mswin, :mingw, :x64_mingw]
   gem "puppet-module-win-dev-r#{minor_version}", '~> 0.4',       require: false, platforms: [:mswin, :mingw, :x64_mingw]
+  gem "puppet-lint-trailing_newline-check", '~> 1.1.0',          require: false
```
Comment (frizop) [inline Gemfile:31]: I don't see a .sync.yml along with this?
```
Gemfile:
  optional:
    ':development':
      - gem: 'puppet-lint-trailing_newline-check'
```
Would be the preference

Comment (frizop) [reply Gemfile:31]: Then you can just run `pdk update` and it'll go and generate all of those gemfile changes

**Review by frizop — CHANGES_REQUESTED** [review]

**Review by frizop — APPROVED** [review]
Looks good!

**Pull request merged**